### ISSUE TITLE
Bug: program.body is not iterable

### ISSUE THREAD

**Issue by draylegend** [azat-io/eslint-plugin-perfectionist#391]
I'm using PhpStorm 2024.3, build #PS-243.21565.202, built on November 13, 2024

### Describe the bug

```js
TypeError: program.body is not iterable
Occurred while linting /home/dl/my-app/apps/app/src/app/app.component.ts/1_inline-template-app.component.ts-1.component.html:1
Rule: "perfectionist/sort-modules"

TypeError: program.body is not iterable
Occurred while linting /home/dl/my-app/apps/app/src/app/app.component.ts/1_inline-template-app.component.ts-1.component.html:1
Rule: "perfectionist/sort-modules"
at analyzeModule (/home/dl/my-app/node_modules/eslint-plugin-perfectionist/dist/rules/sort-modules.js:180:28)
at Program (/home/dl/my-app/node_modules/eslint-plugin-perfectionist/dist/rules/sort-modules.js:159:9)
at ruleErrorHandler (/home/dl/my-app/node_modules/eslint/lib/linter/linter.js:1084:48)
at /home/dl/my-app/node_modules/eslint/lib/linter/safe-emitter.js:45:58
at Array.forEach (<anonymous>)
  at Object.emit (/home/dl/my-app/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
  at NodeEventGenerator.applySelector
  (/home/dl/my-app/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
  at NodeEventGenerator.applySelectors
  (/home/dl/my-app/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
  at NodeEventGenerator.enterNode
  (/home/dl/my-app/node_modules/eslint/lib/linter/node-event-generator.js:337:14)
  at runRules (/home/dl/my-app/node_modules/eslint/lib/linter/linter.js:1128:40)
  Process finished with exit code -1
```

### Code example

```ts
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'app-root',
  imports: [RouterOutlet],
  template: `<router-outlet />`,
})
export class AppComponent {}
```

### Repo

https://github.com/draylegend/perfectionist-bugs

### ESLint version

9.15.0

### ESLint Plugin Perfectionist version

4.0.3

### Additional comments

but perfectionist@3.9.1 does work with eslint@9.14.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @draylegend, I can't reproduce the problem on my end. Could you please provide a repository with your problem, or your ESLint configuration?

**Comment by azat-io** [comment]
For some reason I'm still unable to reproduce the problem.

For some reason I'm still unable to reproduce the problem. I ran `pnpm exec eslint .` command.

```
pnpm exec eslint .

/Users/azat/Developer/perfectionist-bugs/apps/app/src/app/app.component.ts
  6:3  error  Expected "changeDetection" to come before "selector"  perfectionist/sort-objects

/Users/azat/Developer/perfectionist-bugs/apps/app/src/app/app.config.server.ts
  1:34  error  Expected "ApplicationConfig" to come before "mergeApplicationConfig"  perfectionist/sort-named-imports
  4:1   error  Missed spacing between "@angular/ssr" and "./app.config" imports      perfectionist/sort-imports

/Users/azat/Developer/perfectionist-bugs/apps/app/src/app/app.config.ts
  8:1  error  Expected "@angular/platform-browser" (external) to come before "./app.routes" (sibling)  perfectionist/sort-imports

/Users/azat/Developer/perfectionist-bugs/apps/app/src/main.server.ts
  2:1  error  Missed spacing between "@angular/platform-browser" and "./app/app.component" imports  perfectionist/sort-imports

/Users/azat/Developer/perfectionist-bugs/apps/app/src/main.ts
  2:1  error  Missed spacing between "@angular/platform-browser" and "./app/app.config" imports  perfectionist/sort-imports
  3:1  error  Expected "./app/app.component" to come before "./app/app.config"                   perfectionist/sort-imports

/Users/azat/Developer/perfectionist-bugs/apps/app/src/server.ts
  35:5  error  Expected "index" to come before "maxAge"  perfectionist/sort-objects

✖ 8 problems (8 errors, 0 warnings)
  8 errors and 0 warnings potentially fixable with the `--fix` option.
```

The only thing is that I had to remove the `@angular-eslint/template` rule. ESLint did not see it. Can you tell me if I need to install something to make it work?

**Comment by azat-io** [comment]
@draylegend Can you tell me how to reproduce the problem in your repository?

**Comment by draylegend** [comment]
@azat-io, repo changed

**Comment by azat-io** [comment]
@draylegend Can you tell me how you run ESLint? Can you tell me step by step?

I have re-cloned your repository. And everything seems to be working correctly.

```
pnpm exec eslint .

/Users/azat/Developer/perfectionist-bugs/apps/app/src/app/app.component.ts
  2:10  error  'RouterModule' is defined but never used        @typescript-eslint/no-unused-vars
  2:10  error  'RouterModule' is defined but never used        unused-imports/no-unused-imports
  6:3   error  Expected "imports" to come before "standalone"  perfectionist/sort-objects

✖ 3 problems (3 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by draylegend** [comment]
@azat-io did you try with bun?

**Comment by azat-io** [comment]
@draylegend The same result with Bun:

```
bun x eslint .

/Users/azat/Developer/perfectionist-bugs/apps/app/src/app/app.component.ts
  2:10  error  'RouterModule' is defined but never used        @typescript-eslint/no-unused-vars
  2:10  error  'RouterModule' is defined but never used        unused-imports/no-unused-imports
  6:3   error  Expected "imports" to come before "standalone"  perfectionist/sort-objects

✖ 3 problems (3 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

I have a few questions:

1. What is `@angular-eslint/template/attributes-order`? I was commenting out this rule because you do not have the appropriate ESLint plugin for Angular installed.

2. Is your package manager Bun? Could you push the lockfile to repo?

3. What version of Node.js do you use?

4. How do you run ESLint?

**Comment by draylegend** [comment]
1. The corresponding plugin is installed via `angular-eslint`
  ![image](https://github.com/user-attachments/assets/b7c650a9-3c16-43eb-9040-959d70af7cad)
1. [Already in repo](https://github.com/draylegend/perfectionist-bugs/blob/main/bun.lockb)
1. I don't use nodejs at all. Bun is currently my primary JS runtime.
1. PhpStorm supports eslint fixes
  ![image](https://github.com/user-attachments/assets/146efe51-d8ee-4a0d-8324-dfea7ec9927b)

**Comment by azat-io** [comment]
I was able to reproduce the problem. It occurs when linting HTML files.

<img width="1403" alt="Screenshot 2024-11-26 at 12 11 14 AM" src="https://github.com/user-attachments/assets/2712f444-848d-4526-807d-5e8ad2827bcf">

**Comment by hugop95** [comment]
I have managed to reproduce the error: the ESLint configuration is missing this:
<img width="340" alt="Screenshot 2024-11-25 at 22 14 47" src="https://github.com/user-attachments/assets/135a00e7-44f7-460c-8f5e-08f3514189d5">

The issue is due to `angular-eslint` rules with inline templates:
![image](https://github.com/user-attachments/assets/ebacb088-8f6f-4974-91bd-4fda88f3aeb9)

**Comment by hugop95** [comment]
@azat-io Looks like we found the same thing!

**Comment by hugop95** [comment]
@azat-io Perfect!

**Comment by azat-io** [comment]
@draylegend Released in [v4.1.1](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.1.1).

**Comment by draylegend** [comment]
works as expected! What a brilliant job! Thanks @azat-io and @hugop95 for the amazing support and reaction! <3
